### PR TITLE
feat(wallet-dashboard): style migration flow

### LINKED ISSUE #4273: Style migration flow
Style migration flow, ignore warning info boxes for now, they will be added later on when we manage technical edge cases
The assets should be grouped in the exact same way as they are in https://github.com/iotaledger/iota/issues/4271

Depends on https://github.com/iotaledger/iota/issues/4271

Image

### PR DESCRIPTION
**Pull request by evavirseda**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/4273

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                                                />
+                                            )}
+                                        />
+                                    </div>
+                                </Collapsible>
+                                <Panel hasBorder>
+                                    <div className="flex flex-col gap-y-sm p-md">
+                                        <KeyValueInfo
+                                            keyText="Legacy storage deposit"
+                                            value={totalIotaAmountFormatted || '-'}
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]: will this be done later?

**Review by cpl121 — CHANGES_REQUESTED** [review]
conflicts 🙏🏼

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx (hunk 1 above)
Comment (evavirseda) [reply apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]: no, sorry, I misunderstood the function I used... I'll fix it now :see_no_evil:

File: apps/wallet-dashboard/lib/utils/index.ts [hunk 2]
```diff
@@ -8,3 +8,4 @@ export * from './timelock';
 export * from './migration';
 export * from './transaction';
 export * from './growthbook';
+export * from './migration';
```
Comment (brancoder) [inline apps/wallet-dashboard/lib/utils/index.ts]: we should remove this duplicate import

**Review by brancoder — CHANGES_REQUESTED** [review]
There seams to be an issue in the MigrationDialog where user cant scroll to see the last item in the list.
image

File: apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts [hunk 3]
```diff
@@ -76,10 +78,19 @@ export function summarizeMigratableObjectValues({
 
         totalIotaAmount += BigInt(outputObjectFields.balance);
         totalNativeTokens += parseInt(outputObjectFields.native_tokens.fields.size);
-        totalIotaAmount += extractStorageDepositReturnAmount(outputObjectFields, address) || 0n;
+
+        const storageDeposit = extractStorageDepositReturnAmount(outputObjectFields, address);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts]: `extractStorageDepositReturnAmount` checks that the return address is the same as you, we should create a new function that does the opposite

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    open,
+    setOpen,
+    isTimelocked,
+}: MigrationDialogProps): JSX.Element {
+    const account = useCurrentAccount();
+    const { addNotification } = useNotifications();
+    const {
+        data: migrateData,
+        isPending,
+        isError,
```
Comment (marc2332) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]:
```suggestion
        isError: isMigrationError,
```

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    } = useMigrationTransaction(account?.address || '', basicOutputObjects, nftOutputObjects);
+
+    const {
+        data: resolvedObjects = [],
+        isLoading,
+        error: isErrored,
```
Comment (marc2332) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]:
```suggestion
        error: isGroupedMigrationError,
```

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    onSuccess,
+    open,
+    setOpen,
+    isTimelocked,
+}: MigrationDialogProps): JSX.Element {
+    const account = useCurrentAccount();
+    const { addNotification } = useNotifications();
+    const {
+        data: migrateData,
+        isPending,
```
Comment (marc2332) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]:
```suggestion
        isPending: isMigrationPending,
```

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+import { DialogLayout, DialogLayoutBody, DialogLayoutFooter } from './layout';
+import { MigrationObjectDetailsCard } from '../migration/migration-object-details-card';
+import { Collapsible, useFormatCoin } from '@iota/core';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { summarizeMigratableObjectValues } from '@/lib/utils';
+
+interface MigrationDialogProps {
+    basicOutputObjects: IotaObjectData[] | undefined;
+    nftOutputObjects: IotaObjectData[] | undefined;
+    onSuccess?: (digest: string) => void;
```
Comment (marc2332) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]: Why is `onSuccess` optional? Its always passed

Comment (marc2332) [reply apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]: If you make it required, make sure to remove the IF you have below

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/hooks/useResolveStardustGroupedObjects.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    NativeToken,
+}
+
+interface ResolvedStartustObject {
+    type: StardustObjectType;
+    title: string;
+    expirationMs: number;
+    object: IotaObjectData;
+}
+export function useResolveStardustGroupedObjects(
```
Comment (cpl121) [inline apps/wallet-dashboard/hooks/useResolveStardustGroupedObjects.ts]: Where do we use this function?

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+
+    async function handleMigrate(): Promise<void> {
+        if (!migrateData) return;
+        signAndExecuteTransaction(
+            {
+                transaction: migrateData.transaction,
+            },
+            {
+                onSuccess: (tx) => {
+                    onSuccess(tx.digest);
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx:84]: After this, we could show the tx details dialog. We can do it in another issue

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+
+    const { mutateAsync: signAndExecuteTransaction, isPending: isSendingTransaction } =
+        useSignAndExecuteTransaction();
+    const { totalNonOwnedStorageDepositReturnAmount } = summarizeMigratableObjectValues({
+        basicOutputs: basicOutputObjects,
+        nftOutputs: nftOutputObjects,
+        address: account?.address || '',
+    });
+
+    const [gasFee, gasFeesymbol] = useFormatCoin(migrateData?.gasBudget, IOTA_TYPE_ARG);
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]:
```suggestion
    const [gasFee, gasFeeSymbol] = useFormatCoin(migrateData?.gasBudget, IOTA_TYPE_ARG);
```

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    const [
+        totaltotalStorageDepositReturnAmountFormatted,
+        totaltotalStorageDepositReturnAmountSymbol,
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx]:
```suggestion
        totalStorageDepositReturnAmountFormatted,
        totalStorageDepositReturnAmountSymbol,
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx (hunk 9 above)
Comment (begonaalvarezd) [reply apps/wallet-dashboard/components/Dialogs/MigrationDialog.tsx:84]: Thats another issue to speed up this PR https://github.com/iotaledger/iota/issues/4508

File: apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts [hunk 12]
```diff
@@ -52,6 +52,7 @@ interface MigratableObjectsData {
     totalNativeTokens: number;
     totalVisualAssets: number;
     totalIotaAmount: bigint;
+    totalNonOwnedStorageDepositReturnAmount: bigint;
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts]:
```suggestion
    totalNotOwnedStorageDepositReturnAmount: bigint;
```

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
looks very good, just a nit

File: apps/wallet-dashboard/hooks/useResolveStardustGroupedObjects.ts (hunk 8 above)
Comment (evavirseda) [reply apps/wallet-dashboard/hooks/useResolveStardustGroupedObjects.ts]: debris after merge :pray:

**Review by cpl121 — APPROVED** [review]

**Review by begonaalvarezd — APPROVED** [review]

**Review by marc2332 — APPROVED** [review]

**Review by brancoder — APPROVED** [review]
LGTM 🎊

**Pull request merged**